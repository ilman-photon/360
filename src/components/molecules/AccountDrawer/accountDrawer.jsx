import { Box, Button, IconButton, Slide, Stack } from "@mui/material";
import SwipeableDrawer from "@mui/material/Drawer";
import List from "@mui/material/List";
import ListItem from "@mui/material/ListItem";
import ListItemButton from "@mui/material/ListItemButton";
import ListItemIcon from "@mui/material/ListItemIcon";
import ListItemText from "@mui/material/ListItemText";
import KeyboardArrowRightIcon from "@mui/icons-material/KeyboardArrowRight";
import KeyboardArrowLeftIcon from "@mui/icons-material/KeyboardArrowLeft";
import styles from "./accountDrawer.module.scss";
import { useRouter } from "next/router";
import { useState, useEffect } from "react";
import AutoAwesomeMosaicOutlinedIcon from "@mui/icons-material/AutoAwesomeMosaicOutlined";
import CalendarTodayOutlinedIcon from "@mui/icons-material/CalendarTodayOutlined";
import CreateNewFolderOutlinedIcon from "@mui/icons-material/CreateNewFolderOutlined";
import DescriptionOutlinedIcon from "@mui/icons-material/DescriptionOutlined";
import CloseOutlinedIcon from "@mui/icons-material/CloseOutlined";

export const AccountDrawer = ({
  opened = false,
  onClose = () => {
    // This is intended
  },
  onLogoutClicked = () => {
    // This is intended
  },
  menus = [
    {
      label: "Dashboard",
      value: "dashboard",
      icon: <AutoAwesomeMosaicOutlinedIcon />,
    },
    {
      label: "Appointments",
      value: "appointments",
      icon: <CalendarTodayOutlinedIcon />,
    },
    {
      label: "Medical Record",
      value: "medical",
      icon: <CreateNewFolderOutlinedIcon />,
    },
    {
      label: "Documents",
      value: "documents",
      icon: <DescriptionOutlinedIcon />,
    },
  ],
  linkObject = {
    dashboard: [],
    appointments: [],
    medical: [
      {
        label: "Care Plan",
        href: "/patient/account/medical-record/test-lab-result",
      },
      {
        label: "Prescriptions",
        href: "/patient/account/medical-record/test-lab-result",
      },
      {
        label: "Test & Lab Results",
        href: "/patient/account/medical-record/test-lab-result",
      },
    ],
    documents: [
      {
        label: "Intake Forms",
        href: "/patient/account/documents?type=intake-forms",
      },
      {
        label: "Insurance Documents",
        href: "/patient/account/documents?type=insurance-documents",
      },
      {
        label: "Health Record",
        href: "/patient/account/documents?type=health-record",
      },
    ],
  },
}) => {
  const router = useRouter();
  const [activeMenu, setActiveMenu] = useState(null);
  const [sidebarLinks, setSidebarLinks] = useState(null);

  const getMenuTitle = () => {
    if (activeMenu) {
      const found = menus.find((v) => v.value === activeMenu);
      if (found) return found.label;
    }
  };

  useEffect(() => {
    setSidebarLinks(linkObject[activeMenu]);
  }, [activeMenu]);

  const drawerContent = () => (
    <Stack
      sx={{ width: "100vw", padding: "16px", flex: 1 }}
      role="presentation"
    >
      <IconButton sx={{ alignSelf: "end" }} onClick={onClose}>
        <CloseOutlinedIcon />
      </IconButton>
      <List
        sx={{
          flex: 1,
          ".MuiListItem-root": { borderBottomWidth: 1, borderColor: "#F3F3F3" },
        }}
      >
        <Slide direction="left" in={activeMenu} mountOnEnter unmountOnExit>
          <ListItemButton sx={{ background: "#F4F4F4" }}>
            <ListItemIcon sx={{ placeContent: "start" }}>
              <KeyboardArrowLeftIcon />
            </ListItemIcon>
            <ListItemText
              primary={getMenuTitle()}
              onClick={() => setActiveMenu(null)}
            />
          </ListItemButton>
        </Slide>

        <Slide direction="left" in={!sidebarLinks} mountOnEnter unmountOnExit>
          <div>
            {!sidebarLinks &&
              menus.map((menu, idx) => (
                <ListItem
                  key={idx}
                  className={styles.listItemDrawer}
                  sx={{ borderBottomWidth: 1 }}
                >
                  <ListItemButton
                    data-testid="outer-menu-nav-close"
                    onClick={() => setActiveMenu(menu.value)}
                  >
                    <>
                      {menu.icon}
                      <ListItemText primary={menu.label} sx={{ ml: 1 }} />
                    </>
                  </ListItemButton>
                </ListItem>
              ))}
          </div>
        </Slide>

        <Slide direction="left" in={!!sidebarLinks} mountOnEnter unmountOnExit>
          <div>
            {sidebarLinks &&
              sidebarLinks.map((link, idx) => (
                <ListItem key={idx} className={styles.listItemDrawer}>
                  <ListItemButton
                    data-testid="user-menu-nav-close"
                    onClick={async () => {
                      await router.push(link.href);
                      onClose();
                    }}
                  >
                    <>
                      <ListItemText primary={link.label} />
                      <ListItemIcon sx={{ placeContent: "end" }}>
                        <KeyboardArrowRightIcon />
                      </ListItemIcon>
                    </>
                  </ListItemButton>
                </ListItem>
              ))}
          </div>
        </Slide>
      </List>

      <Button
        sx={{
          color: "black",
          display: "flex",
          flexDirection: "row",
          justifyContent: "center",
          alignItems: "center",
          padding: "8px 10px 8px 16px",
          width: "216px",
          height: "36px",
          border: "1px solid #000000",
          borderRadius: "28px",
          margin: "auto",
        }}
      >
        LOG OUT
      </Button>
    </Stack>
  );

  return (
    <>
      <SwipeableDrawer
        anchor="right"
        open={opened}
        onClose={onClose}
        sx={{ width: "70%" }}
      >
        {drawerContent()}
      </SwipeableDrawer>
    </>
  );
};

export default AccountDrawer;
