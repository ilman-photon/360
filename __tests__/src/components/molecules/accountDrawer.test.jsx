import { fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import AccountDrawer from "../../../../src/components/molecules/AccountDrawer/accountDrawer";

window.scrollTo = jest.fn()

describe("AccountDrawer Components", () => {
  let container;
  beforeEach(() => {
    container = render(<AccountDrawer sidebarLinks = {[
        { label: "Profile Information", href: "/patient/account/profile-info" },
        { label: "Financial Information", href: "#" },
        { label: "Toggle accounts", href: "#" },
        { label: "Merge accounts", href: "#" },
        { label: "Prescriptions", href: "#" },
        { label: "Insurance documents", href: "/patient/account/insurance-info" },
        { label: "Multi factor authentication", href: "#" },
      ]}
      opened={true}
      onClose={jest.fn()}
    />);
  });

  it("AccountDrawer render", () => {
    expect(container).toMatchSnapshot();

    setTimeout(() => {
      expect("Profile Information").toBeVisible() 
      expect("Financial Information").toBeVisible()
      expect("Toggle accounts").toBeVisible() 
      expect("Merge accounts").toBeVisible()
      expect("Prescriptions").toBeVisible() 
      expect("Insurance documents").toBeVisible()
      expect("Multi factor authentication").toBeVisible()
      expect("LOG OUT").toBeVisible()
    }, 500);
  });

});
