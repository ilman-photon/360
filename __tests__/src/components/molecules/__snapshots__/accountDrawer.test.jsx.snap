// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDrawer Components AccountDrawer render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDrawer-root MuiDrawer-modal MuiModal-root css-n6x5hq-MuiModal-root-MuiDrawer-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation16 MuiDrawer-paper MuiDrawer-paperAnchorRight css-1160xiw-MuiPaper-root-MuiDrawer-paper"
        tabindex="-1"
      >
        <div
          class="css-roem2o-MuiStack-root"
          role="presentation"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-bzwnf2-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <ul
            class="MuiList-root MuiList-padding css-1xob6qv-MuiList-root"
          >
            <div
              style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
            >
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding listItemDrawer css-84ly9z-MuiListItem-root"
              >
                <div
                  class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                  data-testid="outer-menu-nav-close"
                  role="button"
                  tabindex="0"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="AutoAwesomeMosaicOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 5v14c0 1.1.89 2 2 2h6V3H5c-1.11 0-2 .9-2 2zm6 14H5V5h4v14zM19 3h-6v8h8V5c0-1.1-.9-2-2-2zm0 6h-4V5h4v4zm-6 12h6c1.1 0 2-.9 2-2v-6h-8v8zm2-6h4v4h-4v-4z"
                    />
                  </svg>
                  <div
                    class="MuiListItemText-root css-5t57t0-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      Dashboard
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </div>
              </li>
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding listItemDrawer css-84ly9z-MuiListItem-root"
              >
                <div
                  class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                  data-testid="outer-menu-nav-close"
                  role="button"
                  tabindex="0"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CalendarTodayOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V10h16v11zm0-13H4V5h16v3z"
                    />
                  </svg>
                  <div
                    class="MuiListItemText-root css-5t57t0-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      Appointments
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </div>
              </li>
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding listItemDrawer css-84ly9z-MuiListItem-root"
              >
                <div
                  class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                  data-testid="outer-menu-nav-close"
                  role="button"
                  tabindex="0"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CreateNewFolderOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 6h-8l-2-2H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm0 12H4V6h5.17l2 2H20v10zm-8-4h2v2h2v-2h2v-2h-2v-2h-2v2h-2z"
                    />
                  </svg>
                  <div
                    class="MuiListItemText-root css-5t57t0-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      Medical Record
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </div>
              </li>
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding listItemDrawer css-84ly9z-MuiListItem-root"
              >
                <div
                  class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                  data-testid="outer-menu-nav-close"
                  role="button"
                  tabindex="0"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="DescriptionOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8 16h8v2H8zm0-4h8v2H8zm6-10H6c-1.1 0-2 .9-2 2v16c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm4 18H6V4h7v5h5v11z"
                    />
                  </svg>
                  <div
                    class="MuiListItemText-root css-5t57t0-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      Documents
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </div>
              </li>
            </div>
          </ul>
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-19ty54a-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            LOG OUT
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AccountDrawer Components AccountDrawer render 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDrawer-root MuiDrawer-modal MuiModal-root css-n6x5hq-MuiModal-root-MuiDrawer-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation16 MuiDrawer-paper MuiDrawer-paperAnchorRight css-1160xiw-MuiPaper-root-MuiDrawer-paper"
        tabindex="-1"
      >
        <div
          class="css-roem2o-MuiStack-root"
          role="presentation"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-bzwnf2-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <ul
            class="MuiList-root MuiList-padding css-1xob6qv-MuiList-root"
          >
            <div
              style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
            >
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding listItemDrawer css-84ly9z-MuiListItem-root"
              >
                <div
                  class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                  data-testid="outer-menu-nav-close"
                  role="button"
                  tabindex="0"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="AutoAwesomeMosaicOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 5v14c0 1.1.89 2 2 2h6V3H5c-1.11 0-2 .9-2 2zm6 14H5V5h4v14zM19 3h-6v8h8V5c0-1.1-.9-2-2-2zm0 6h-4V5h4v4zm-6 12h6c1.1 0 2-.9 2-2v-6h-8v8zm2-6h4v4h-4v-4z"
                    />
                  </svg>
                  <div
                    class="MuiListItemText-root css-5t57t0-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      Dashboard
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </div>
              </li>
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding listItemDrawer css-84ly9z-MuiListItem-root"
              >
                <div
                  class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                  data-testid="outer-menu-nav-close"
                  role="button"
                  tabindex="0"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CalendarTodayOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V10h16v11zm0-13H4V5h16v3z"
                    />
                  </svg>
                  <div
                    class="MuiListItemText-root css-5t57t0-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      Appointments
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </div>
              </li>
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding listItemDrawer css-84ly9z-MuiListItem-root"
              >
                <div
                  class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                  data-testid="outer-menu-nav-close"
                  role="button"
                  tabindex="0"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CreateNewFolderOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 6h-8l-2-2H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm0 12H4V6h5.17l2 2H20v10zm-8-4h2v2h2v-2h2v-2h-2v-2h-2v2h-2z"
                    />
                  </svg>
                  <div
                    class="MuiListItemText-root css-5t57t0-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      Medical Record
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </div>
              </li>
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding listItemDrawer css-84ly9z-MuiListItem-root"
              >
                <div
                  class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                  data-testid="outer-menu-nav-close"
                  role="button"
                  tabindex="0"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="DescriptionOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8 16h8v2H8zm0-4h8v2H8zm6-10H6c-1.1 0-2 .9-2 2v16c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm4 18H6V4h7v5h5v11z"
                    />
                  </svg>
                  <div
                    class="MuiListItemText-root css-5t57t0-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      Documents
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </div>
              </li>
            </div>
          </ul>
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-19ty54a-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            LOG OUT
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
